blocks/bricks/sequences.py

```diff
@@ -45,6 +46,32 @@ def apply_inputs(self):
     def apply_outputs(self):
         return self.application_methods[-1].outputs
 
+    def __delitem__(self, index):
+        application_method = self.application_methods[index]
+        brick = application_method.brick
+        try:
+            del self.application_methods[index]
+        finally:
+            apps_with_brick = [a for a in self.application_methods
```

Comment:
This look weird to me. Why do you want to modify the brick if an exception was raised on `del self.application_methods[index]`?